azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -143,17 +147,46 @@ def create_from_edge_environment(cls):
             workload_uri = os.environ["IOTEDGE_WORKLOADURI"]
             api_version = os.environ["IOTEDGE_APIVERSION"]
         except KeyError:
-            raise auth.IoTEdgeError("IoT Edge container not configured correctly")
-
-        authentication_provider = auth.IoTEdgeAuthenticationProvider(
-            hostname=hostname,
-            device_id=device_id,
-            module_id=module_id,
-            gateway_hostname=gateway_hostname,
-            module_generation_id=module_generation_id,
-            workload_uri=workload_uri,
-            api_version=api_version,
-        )
+            # As a fallback, try the Edge local dev variables for debugging.
+            # These variables are set by VS/VS Code in order to allow debugging
+            # of Edge application code in a non-Edge dev environment.
+            try:
+                connection_string = os.environ["EdgeHubConnectionString"]
+                ca_cert_filepath = os.environ["EdgeModuleCACertificateFile"]
+            except KeyError:
+                # TODO: consider using a different error here. (OSError?)
+                raise auth.IoTEdgeError("IoT Edge environment not configured correctly")
+
+            # Read the certificate file to pass it on as a string
+            try:
+                with io.open(ca_cert_filepath, mode="r") as ca_cert_file:
```

Comment:
>io.open [](start = 21, length = 7)

Did you try using just `open` ?

Reply:
`io.open` is recommended for code that has to work between python2/3.

---
In reply to: [300470387](https://github.com/Azure/azure-iot-sdk-python-preview/pull/117#discussion_r300470387) [](ancestors = 300470387)